File: Ruby/Civitas/lib/sorpresa_pagar_cobrar.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
  class SorpresaPagarCobrar < Sorpresa
    def initialize(valor, texto)
      super()
      @valor=valor
      @texto=texto
      
      #puts "Sorpresa PagarCobrar creada.\n"
    end
    
    # Método sobrescrito
    def aplicar_a_jugador(actual, todos)
      if(jugador_correcto(actual, todos))
        todos[actual].modificar_saldo(self.valor)
      end
    end

    def to_s
       "\n\t ¡SORPRESA! \n
       \n Has recibido: 
       \n #{@texto} \n"
    end
  end
end

File: Java/P4/Civitas/src/civitas/SorpresaPagarCobrar.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author arturo
 */
public class SorpresaPagarCobrar extends Sorpresa{
    private int valor;
    
    SorpresaPagarCobrar(int valor, String texto){
        super(texto);
        this.valor=valor;
    }
    
    @Override
    public void aplicarAJugador(int actual, ArrayList<Jugador> todos){
        if(this.jugadorCorrecto(actual, todos)){
            todos.get(actual).modificarSaldo(this.valor);
        }
    }
    
    @Override   
    public String toString(){        
        return "Sorpresa "+super.getTexto()+", valor que recibes/pagas= "+valor+"\n";
    }
}